ambry-server/src/main/java/com/github/ambry/server/mysql/AccountStatsMySqlStore.java

```diff
@@ -125,21 +137,98 @@ public void publish(StatsWrapper statsWrapper) {
               .computeIfAbsent("Partition[" + partitionId + "]", k -> new StatsSnapshot((long) 0, new HashMap<>()));
           StatsSnapshot accountSnapshot = partitionSnapshot.getSubMap()
               .computeIfAbsent("A[" + accountId + "]", k -> new StatsSnapshot((long) 0, new HashMap<>()));
           accountSnapshot.getSubMap().put("C[" + containerId + "]", new StatsSnapshot(storageUsage, null));
+          timestamp.set(Math.max(timestamp.get(), updatedAtMs));
         });
 
     hostSnapshot.updateValue();
-    return hostSnapshot;
+    return new StatsWrapper(
+        new StatsHeader(StatsHeader.StatsDescription.STORED_DATA_SIZE, timestamp.get(), partitionSubMap.size(),
+            partitionSubMap.size(), null), hostSnapshot);
+  }
+
+  /**
+   * Store the aggregated account stats in {@link StatsSnapshot} to mysql database.
+   * @param snapshot The aggregated account stats snapshot.
+   */
+  @Override
+  public void storeAggregatedStats(StatsSnapshot snapshot) throws SQLException {
+    int batchSize = 0;
+    long startTimeMs = System.currentTimeMillis();
+    for (Map.Entry<String, StatsSnapshot> accountMapEntry : snapshot.getSubMap().entrySet()) {
+      String accountIdKey = accountMapEntry.getKey();
+      short accountId = Short.valueOf(accountIdKey.substring(2, accountIdKey.length() - 1));
+      StatsSnapshot containerStatsSnapshot = accountMapEntry.getValue();
+      for (Map.Entry<String, StatsSnapshot> currContainerMapEntry : containerStatsSnapshot.getSubMap().entrySet()) {
+        String containerIdKey = currContainerMapEntry.getKey();
+        short containerId = Short.valueOf(containerIdKey.substring(2, containerIdKey.length() - 1));
+        long currStorageUsage = currContainerMapEntry.getValue().getValue();
+        aggregatedaccountReportsDao.updateStorageUsage(accountId, containerId, currStorageUsage);
+        batchSize++;
+      }
+    }
+    storeMetrics.aggregatedPublishTimeMs.update(System.currentTimeMillis() - startTimeMs);
+    storeMetrics.aggregatedBatchSize.update(batchSize);
+  }
+
+  /**
+   * Query mysql database to get all the aggregated container storage usage for given {@code clusterName} and construct
+   * a map from those data. The map is structured as such:
+   * <p>Outer map's key is the string format of account id, inner map's key is the string format of container id and the
+   * value of the inner map is the storage usage of the container.</p>
+   * @param clusterName the clusterName.
+   * @return A map that represents container storage usage.
+   * @throws Exception
+   */
+  @Override
+  public Map<String, Map<String, Long>> queryAggregatedStats(String clusterName) throws Exception {
+    Map<String, Map<String, Long>> result = new HashMap<>();
+    aggregatedaccountReportsDao.queryContainerUsageForCluster(clusterName, (accountId, containerId, storageUsage) -> {
+      result.computeIfAbsent(String.valueOf(accountId), k -> new HashMap<>())
```

Comment:
Do we want to use the latest updated storageUsage value or just the last one in order?

Reply:
For aggregated stats, there is only one storage usage. When there is a new value, it will override the old ones in database.